packages/esm-patient-clinical-view-app/src/clinical-view-form/clinical-view-form.component.tsx

```diff
@@ -1,36 +1,62 @@
-import React, { useState, useCallback, useMemo } from 'react';
+import React, { useState, useCallback, useMemo, useReducer } from 'react';
 import Search from 'carbon-components-react/es/components/Search';
 import Checkbox from 'carbon-components-react/es/components/Checkbox';
 import Button from 'carbon-components-react/es/components/Button';
 import styles from './clinical-view-form.component.scss';
 import debounce from 'lodash-es/debounce';
 import { useTranslation } from 'react-i18next';
-import { detach, temporaryConfigStore, TemporaryConfigStore, useConfig } from '@openmrs/esm-framework';
+import { detach, temporaryConfigStore, TemporaryConfigStore } from '@openmrs/esm-framework';
 import { useClinicalView } from '../store';
 import isEmpty from 'lodash-es/isEmpty';
 import cloneDeep from 'lodash-es/cloneDeep';
 import set from 'lodash-es/set';
+import { StructuredListSkeleton } from 'carbon-components-react/es/components/StructuredList';
+
+enum ActionTypes {
+  pending = 'pending',
+  resolved = 'resolved',
+}
+interface Pending {
+  type: ActionTypes.pending;
+}
+interface Resolved {
+  type: ActionTypes.resolved;
+}
+
+interface Status {
+  status: 'pending' | 'resolved';
+}
+
+type Action = Pending | Resolved;
+
+function reducer(state: Status, action: Action): Status {
+  return { status: action.type };
+}
```

Comment:
In the current form the reducer can just be replaced with a simple `useState`, perhaps even with a `boolean`:

```ts
const [isPending, setIsPending] = useState(true);
```